system/handlers/formbuilder/item-types/CheckboxList.cfc

```diff
@@ -17,41 +17,45 @@ component {
 	}
 
 	private string function renderResponse( event, rc, prc, args={} ) {
-		var responses = ListToArray( args.response );
-		var values    = ListToArray( args.itemConfiguration.values, chr(13) & chr(10) );
-		var labels    = ListToArray( args.itemConfiguration.labels, chr(13) & chr(10) );
-
-		var renderLabels = [];
-		for ( var response in responses ) {
-			var ix = values.find( response );
-			if ( ix>0 ) {
-				renderLabels.append( labels[ix] ?: values[ix] );
-			}
-		}
-
-		return ArrayToList( renderLabels, ", " );
+		return ArrayToList( _renderResponses( argumentCollection=arguments, useLabel=true ), args.delim ?: "<br>" );
 	}
 
 	private array function renderResponseForExport( event, rc, prc, args={} ) {
-		var responses = ListToArray( args.response.replace( """", "", "all") );
-		var values    = ListToArray( args.itemConfiguration.values, chr(13) & chr(10) );
-		var labels    = ListToArray( args.itemConfiguration.labels, chr(13) & chr(10) );
-		var renderLabels = [];
-		for ( var response in responses ) {
-			var ix = values.find( response );
-			if ( ix>0 ) {
-				renderLabels.append( labels[ix] ?: values[ix] );
-			}
-		}
-		return [ ArrayToList( renderLabels, ", " ) ];
+		return [ ArrayToList( _renderResponses( argumentCollection=arguments, useLabel=true ), ", " ) ];
 	}
 
 	private array function renderV2ResponsesForDb( event, rc, prc, args={} ) {
-
-		return ListToArray( args.response ?: "" );
+		return _renderResponses( argumentCollection=arguments );
 	}
 
 	private string function getQuestionDataType( event, rc, prc, args={} ) {
 		return "shorttext";
 	}
+
+	private array function _renderResponses( event, rc, prc, args={}, boolean useLabel=false ) {
+		var searchResponse = ",#args.response#,";
+
+		if ( isEmptyString( searchResponse ) ) {
+			return [];
+		}
+
+		var itemConfig     = args.itemConfiguration ?: ( args.configuration ?: {} );
+		var matchResponses = [];
+
+		var values = ListToArray( itemConfig.values ?: "", chr(13) & chr(10) );
+		var labels = ListToArray( itemConfig.labels ?: "", chr(13) & chr(10) );
+
+		for ( var i=1; i<=ArrayLen( values ); i++ ) {
+			if ( Find( ",#values[ i ]#,", searchResponse ) ) {
+				var renderResponse = useLabel ? ( labels[ i ] ?: values[ i ] ) : values[ i ];
+
+				ArrayAppend( matchResponses, renderResponse );
+
+				searchResponse = REReplace( Replace( searchResponse, ",#values[ i ]#,", "" ), "^,?(.*?),?$", ",\1," );
```

Comment:
I think this can just be:

```cfc
searchResponse = Replace( searchResponse, ",#values[ i ]#,", "," );
```

i.e. replace it with a single comma, rather than empty string.